### ISSUE TITLE
Add HEALTHCHECK

### ISSUE THREAD

**Issue by jamesmr89** [jacobalberty/unifi-docker#58]
Was just reading over this:

https://blog.newrelic.com/2016/08/24/docker-health-check-instruction/

Seems like something like this would suffice:

curl -k -L --fail https://localhost:8443 || exit 1

**Comment by jacobalberty** [comment]
I'm assuming you mean for travis-ci build tests. Trying to integrate this now, though at the moment it simply causes a build failure for 
```
$ curl -k -L --fail https://127.0.0.1:8443 || exit 1
curl: (35) Unknown SSL protocol error in connection to 127.0.0.1:8443 
```

**Comment by jamesmr89** [comment]
I don't think this has anything to do with travis-ci, this is to be appended to the dockerfile

( I do think this means you'll have to add curl package to the build as well)

https://docs.docker.com/engine/reference/builder/#healthcheck

Something like this appended to dockerfile:

HEALTHCHECK --interval=5m --timeout=3s CMD curl -f https://localhost:8443 || exit 1

Then when doing docker ps you will be able to see the healthy/unhealthy state

**Comment by jacobalberty** [comment]
Yeah I took a closer look and realized its a new dockerfile feature i wasn't familiar with, going to try and get it into travis-ci as well though as it will improve build checks too. Adding it to beta shortly

**Comment by jacobalberty** [comment]
It's merged into beta now, I have a few other changes that I'm working on with beta then I'm pushing all of the changes from beta into all of the other tags, I'll leave this open till I get beta merged back out into master.

**Comment by jcberthon** [comment]
Hi @jacobalberty 

If you are interested, I have already healthcheck (Docker feature) implemented for my Unifi container. See here: https://github.com/jcberthon/unifi-docker/blob/4a902fd89386b473098e42c6f763a497c8500dea/Dockerfile#L61. I've added that at the end of August, so it is pretty stable.

Note that this is most useful when running the container in a Swarm. It can also be of interest for the few doing `docker ps` is order to see if everything is fine. But usually you see it right away when you connect to it ;-)

What can be interesting for you is the use of the different options for the `HEALTHCHECK` instructions.

*Note: On my personal copy image, I have commented out the HEALTHCHECK instruction because my CI builder is still stuck with Docker 1.11. However, you can add/override the healthcheck in the compose file. That's what I'm currently using at home.*

**Comment by jacobalberty** [comment]
I have a healthcheck working in the beta branch now, I'm cleaning up beta to include a few other changes (debian-stretch, apt instead of gdebi and non root user support) to make it easier to support non x86 platforms before I push beta to the other branches.

**Comment by jamesmr89** [comment]
Hi @jcberthon, I'm wondering how well this works for you in swarm mode?  I'm currently running in swarm mode and have to run a docker service scale unifi=0 and then subsequently a docker service scale unifi=1, once i receive a notification that the unfi site is down (About 2x a day).  Are you essentially able to automate this from swarm with the above setup?

**Comment by jacobalberty** [comment]
This is now merged into main branch and building on docker hub, I'll go ahead and close this once the build is complete.

**Comment by jcberthon** [comment]
Hi @jacobalberty 

Sorry for the late reply. This last 2 weeks have been low activity for me. Kids getting sick and I need sleep/rest.

Anyway, I haven't managed to try it in Swarm mode at the moment. The reason is that I only have 1 amd64 "server". My swarm is currently made of 2 Raspberry Pis, but it was for a long time not working until fixed by Docker CE 17.09, and my image of UniFi is not yet working on ARM but I did not find the time to fix it.

How did you solve the storage problem in Swarm? I've been using GlusterFS, but my current services running in Swarm are little demanding in terms of IO. I'm wondering how well Mongo would fare, or if it would be better with another approach.
